spec/requests/users_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe '/Users', type: :request do
+  describe 'Get/index' do
+    before(:example) do
+      get '/users'
+    end
+    it 'renders a successful response' do
+      expect(response).to be_successful
+    end
+    it 'renders the correct template' do
+      expect(response).to render_template(:index)
+    end
+    it 'contains the correct text' do
+      expect(response.body).to include('Here is a list of all users')
+    end
+  end
+
+  describe 'Get/show' do
+    before(:example) do
+      get '/users/1'
+    end
+    it 'renders a successful response' do
+      expect(response).to be_successful
+    end
+    it 'renders the correct template' do
+      expect(response).to render_template(:show)
+    end
+    it 'contains the correct text' do
+      expect(response.body).to include('Here are the detials about User ABC')
+    end
+  end
```

Comment:
- [ ] There are 6 failed tests, kindly fix that.

![image](https://user-images.githubusercontent.com/68796232/231195645-a69d3e31-11eb-41f7-87d0-beda300b7c51.png)

That happened because the show methods in both users and posts controllers search for a user or post with `id = 1` but didn't find any because you didn't add any users or posts before tests.

![image](https://user-images.githubusercontent.com/68796232/231199346-e3299b69-6d6a-4158-ab6e-03d5470ce855.png)

Reply:
hello, @Omar-Muhamad  thank you to review my pull request but as I can see the tests are falling in your machine but all tests are passing in my local machine, could you have any ideas what could be the reason for making this happen?
![image](https://user-images.githubusercontent.com/104406349/231204988-90936781-0d7a-4386-b0a5-77c489d3ebdf.png)